### PR TITLE
Adds serializers and views for website submission API endpoints.

### PR DESCRIPTION
**Pull request by chuckharmston**
r? @ngokevin

### CODE REVIEW

File: mkt/submit/helpers.py [hunk 1]
```diff
@@ -65,9 +68,11 @@ def string_to_translatedfield_value(text):
...
-    lang = guess_language(text)
-    if lang:
-        return {lang: text}
-    return {'en-us': text}
+    guess = guess_language(text)
+    if guess:
+        lang = find_language(guess).lower()
+        if lang:
+            return {lang: text}
+    return {settings.SHORTER_LANGUAGES['en'].lower(): text}
```
Comment (chuckharmston) [inline mkt/submit/helpers.py]: @diox this is where it ended up. Works brilliantly.

File: mkt/submit/helpers.py [hunk 2]
```diff
@@ -65,9 +68,11 @@ def string_to_translatedfield_value(text):
...
-    determined with confidence, will assume en-US.
+    determined with confidence, will assume English.
     """
-    lang = guess_language(text)
-    if lang:
-        return {lang: text}
-    return {'en-us': text}
+    guess = guess_language(text)
+    if guess:
+        lang = find_language(guess).lower()
```
Comment (chuckharmston) [inline mkt/submit/helpers.py]: This is to do the transform from `en` => `en-US`, etc.

File: mkt/websites/views.py [hunk 3]
```diff
@@ -98,3 +100,15 @@ def get(self, request, format=None):
...
+class WebsiteSubmissionView(CORSMixin, MarketplaceView, CreateAPIView):
+    cors_allowed_methods = ['post']
+    authentication_classes = [RestSharedSecretAuthentication,
+                              RestOAuthAuthentication]
+    model = WebsiteSubmission
+    permission_classes = [GroupPermission('Websites', 'Submit')]
+    serializer_class = PublicWebsiteSubmissionSerializer
+
+    def pre_save(self, obj):
+        setattr(obj, 'submitter', self.request.user)
```
Comment (ngokevin) [inline mkt/websites/views.py]: Do I need to send a submitter then?

**Comment by ngokevin** [comment]
r+

File: mkt/websites/views.py (hunk 3 above)
Comment (chuckharmston) [inline mkt/websites/views.py]: Nope, as long as you're authenticated.

**Comment by chuckharmston** [comment]
Thanks! Waiting on tests first, then will merge.

**Comment by chuckharmston** [comment]
Nudging Travis

**Pull request merged**